language-plutus-core/src/Data/Aeson/THReader.hs

```diff
@@ -0,0 +1,15 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Data.Aeson.THReader where
+
+import Data.Aeson
+import Language.Haskell.TH
+import Language.Haskell.TH.Syntax
+import qualified Data.ByteString.Lazy as BSL
+
+readJSONFromFile :: (FromJSON a, Lift a) => String -> Q (TExp a)
+readJSONFromFile name = do
+    addDependentFile name
+    contents <- runIO $ BSL.readFile name
+    case (eitherDecode contents) of
+        Left err -> error err
```

Comment:
I vaguely remember that we want to use `fail` as `Q` has a sensible `MonadFail` instance.

Reply:
Found `error` in some tutorials. Will fix.